webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreService.scala

```diff
@@ -48,32 +49,33 @@ trait ScoreService {
   def getAll(): IOResult[Map[NodeId, GlobalScore]]
   def getGlobalScore(nodeId:  NodeId): IOResult[GlobalScore]
   def getScoreDetails(nodeId: NodeId): IOResult[List[Score]]
-  def clean(): IOResult[Unit]
-  def cleanScore(name:          String): IOResult[Unit]
-  def update(newScores:         Map[NodeId, List[Score]]): IOResult[Unit]
+  def clean()(implicit qc: QueryContext): IOResult[Unit]
+  def cleanScore(name:          String)(implicit qc: QueryContext): IOResult[Unit]
+  def update(newScores:         Map[NodeId, List[Score]])(implicit qc : QueryContext): IOResult[Unit]
   def registerScore(newScoreId: String, displayName: String): IOResult[Unit]
   def getAvailableScore(): IOResult[List[(String, String)]]
-  def init():              IOResult[Unit]
+  def init()(implicit qc : QueryContext):              IOResult[Unit]
   def deleteNodeScore(nodeId: NodeId): IOResult[Unit]
 }
 
-class ScoreServiceImpl(globalScoreRepository: GlobalScoreRepository, scoreRepository: ScoreRepository) extends ScoreService {
+class ScoreServiceImpl(globalScoreRepository: GlobalScoreRepository, scoreRepository: ScoreRepository, nodeFactRepo: NodeFactRepository) extends ScoreService {
   private val cache:      Ref[Map[NodeId, GlobalScore]] = Ref.make(Map.empty[NodeId, GlobalScore]).runNow
   private val scoreCache: Ref[Map[NodeId, List[Score]]] = Ref.make(Map.empty[NodeId, List[Score]]).runNow
 
   private val availableScore: Ref[List[(String, String)]] =
     Ref.make((ComplianceScore.scoreId, "Compliance") :: (SystemUpdateScore.scoreId, "System updates") :: Nil).runNow
 
-  def init(): IOResult[Unit] = {
+  def init()(implicit qc : QueryContext): IOResult[Unit] = {
     for {
+      nodeIds <- nodeFactRepo.getAll().map(_.keySet)
       globalScores <- globalScoreRepository.getAll()
-      _            <- cache.set(globalScores)
+      _            <- cache.set(globalScores.filter(n => nodeIds.contains(n._1)))
```

Comment:
here, `nodeIds` only contains node as seen by the person issuing the query context. I think we want all (active) nodes at that point, so directly using the system query context here, and not as a parameter (even if it's called with that, I don't think it should be a parameter at all).

Reply:
I don't know it gives access to score of all Nodes even those you cannot, like a leak of data to another tenant

Reply:
Sorry i haven't seen it was in init, it should be systemQC always (and it is always used with it)